pulpcore/app/models/content.py

```diff
@@ -58,8 +58,6 @@ def q(self):
         """
         Returns a Q object that represents the model
         """
-        if self.pk:
```

Comment:
I don't know what the performance ramifications of this are, perhaps it should "try" to search by PK first, and if it fails, do the natural key search?

Reply:
I wondered about that too. That seems like a decent option to me, its cleaner than https://github.com/pulp/pulpcore-plugin/pull/29 and seems like the correct behavior in all circumstances.

Reply:
@bmbouter What should we do about this PR?

Reply:
This commit message describes the issue well. It's probably safe to do this. I'm not sure of the performance implications.

Reply:
@bmbouter would this fix the issue? https://github.com/pulp/pulp/compare/master...dralley:uuid#diff-5308662d441ee7c5774295277cba139cL63

Reply:
@dralley This diff seems empty?